File: plugin/octo_beat_plugin/android/src/main/kotlin/com/octo/octo_beat_plugin/core/device/model/TCPConnection.java
package com.octo.octo_beat_plugin.core.device.model;

import com.octo.octo_beat_plugin.core.device._enum.SSLAuthMode;

public class TCPConnection {
    public int id;
    public SSLAuthMode authMode;
    public int serverCaId;
    public int clientCaId;
    public int clientPrivateKeyId;
    public boolean isSSLEnabled;

    public TCPConnection(int id, SSLAuthMode authMode, int serverCaId, int clientCaId, int clientPrivateKeyId) {
        this.id = id;
        this.authMode = authMode;
        this.serverCaId = serverCaId;
        this.clientCaId = clientCaId;
        this.clientPrivateKeyId = clientPrivateKeyId;
    }

    public void paste(TCPConnection connection) {
        this.id = connection.id;
        this.authMode = connection.authMode;
        this.serverCaId = connection.serverCaId;
        this.clientCaId = connection.clientCaId;
        this.clientPrivateKeyId = connection.clientPrivateKeyId;
    }

}

File: plugin/octo_beat_plugin/ios/Classes/core/device/handler/model/TCPConnection.swift
//
//  TCPConnection.swift
//  DXH-ios
//
//  Created by Manh Tran on 11/26/19.
//  Copyright © 2019 software-2. All rights reserved.
//

import Foundation

enum SSLAuthMode {
    case ONE_WAY_AUTH
    case TWO_WAY_AUTH
}
class TCPConnection {
    var id: UInt8!
    var authMode: SSLAuthMode!
    var serverCAId: UInt8!
    var clientCAId: UInt8!
    var clientPrivateKeyId: UInt8!
    var isSSLEnable: Bool!
    
    init(id: UInt8, authMode: SSLAuthMode, serverCAId: UInt8, clientCAId: UInt8, clientPrivateKeyId: UInt8) {
        self.id = id
        self.authMode = authMode
        self.serverCAId = serverCAId
        self.clientCAId = clientCAId
        self.clientPrivateKeyId = clientPrivateKeyId
    }
    
    func paste(connection: TCPConnection){
        self.id = connection.id
        self.authMode = connection.authMode
        self.serverCAId = connection.serverCAId
        self.clientCAId = connection.clientCAId
        self.clientPrivateKeyId = connection.clientPrivateKeyId
    }
}
